internal/controllers/topology/cluster/desired_state.go

```diff
@@ -823,17 +824,32 @@ func computeMachineHealthCheck(healthCheckTarget client.Object, selector *metav1
 	// state of the object won't be different from the current state due to webhook Defaulting.
 	mhc.Default()
 
-	// Carry over ownerReference to the target object.
-	// NOTE: this prevents to the ownerRef to be deleted by server side apply.
-	if current != nil {
-		if ref := getOwnerReferenceFrom(current, healthCheckTarget); ref != nil {
-			mhc.SetOwnerReferences([]metav1.OwnerReference{*ref})
-		}
+	if !isNil(currentHealthCheckTarget) {
+		// If the currentHealthCheckTarget exists we use it to pick up the UID (UID is not set on desired).
+		mhc.SetOwnerReferences([]metav1.OwnerReference{
+			*ownerReferenceTo(currentHealthCheckTarget),
+		})
+	} else {
+		// Otherwise we have to use desiredHealthCheckTarget. UID is then resolved during reconcile state.
+		mhc.SetOwnerReferences([]metav1.OwnerReference{
+			*ownerReferenceTo(desiredHealthCheckTarget),
+		})
 	}
 
 	return mhc
 }
 
+func isNil(i interface{}) bool {
```

Comment:
I intentionally copied this func for now instead of taking on a dependency. We might want to reconsider this later and have a separate nil util kind of thing.

I wonder if there was a change in some Go version as we're recently running more often into the situation that a simple nil check is not enough